### ISSUE TITLE
[@typescript-eslint/no-unsafe-assignment] False positive on any type assignment

### ISSUE THREAD

**Issue by j-a-y-h** [typescript-eslint/typescript-eslint#2289]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-assignment": ["error"]
  }
}
```

```TS
type Value = any;
const a: Value = 5;
const b: Value = a;
```

**Expected Result**
Since `b` is expected to be` any`, the linter should not throw an error.

**Actual Result**
Linter throws an error for unsafe assignment.

**More information**
This occurs with other rules like `@typescript-eslint/no-unsafe-return`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.2` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.9.1` |
| `npm`                              | `6.10.3` |

**Comment by bradzacher** [comment]
Why are you explicitly typing `any` at all?

**Comment by j-a-y-h** [comment]
It's a type alias to track a specific type of input that could be in any shape.

**Comment by j-a-y-h** [comment]
Thanks for the response

**Comment by bradzacher** [comment]
Sorry, I mean why not use the much safer `unknown` type?

**Comment by j-a-y-h** [comment]
That the behavior I want but `unknown` will throw an error if used in a form input.

**Comment by bradzacher** [comment]
Isn't that kind of the point?
You shouldn't be passing an `any` typed thing anywhere, least of all into a place that's expecting a strictly typed value.
That is inherently unsafe and will lead to hard-to-debug runtime errors.

```ts
function myFunc(arg: string) { ... }

const x: any = 1;
myFunc(x); // this should be a compiler error, because it will be a runtime error.
if (typeof y === 'string') {
  myFunc(y); // safe, but no way to enforce a check is done before calling
}
```

OTOH, if you use `unknown`, you will _have_ to do an explicit assertion before hand, ensuring your code is compile time AND runtime safe.

```ts
const y: unknown = 1;
myFunc(y); // compiler error
if (typeof y === 'string') {
  myFunc(y); // safe
}
```

The TL;DR is _do not_ use the type `any` ever.

**Comment by j-a-y-h** [comment]
Thanks for the response.  I'm aware of the dangers of using type any in a project and I would love to stick to the `do not use the type any ever`, but that's not always possible especially when you are dealing with user inputs.  Not to mention stating `do not` use something ever is a pretty strong statement especially when typescriptlang.org explicitly has [use cases](https://www.typescriptlang.org/docs/handbook/basic-types.html#any) for this particular situation on their website.

Having said that, here is a basic example of why I have to use any.  I have a type that is used in form inputs that could be of the strict form value type of `string | number | string[]` or it could be of a JSON type (`type Value = string | number | string[] | {[key: string]: Value}`) when the user is using nested objects.  Going with the second definition would throw an error if the user used the value directly in the form without casting it.  This is the same as using unknown as the user still has to cast it.  `any` allows the user of the library to use it in forms out the box without errors and the type alias allows me to track that input flow from the user.

The only other possible solution I can think of would be to use a generic but I haven't fully assessed how that implementation would work and it might result in the same 'user needs to cast' situation.  Regardless, I still think you should be allowed to assign `any` to `any` since they share the same type and I'm not aware of anything in the typescript language that prohibits you from assigning `any` to `any`.  

I would love to continue using those two rules but until this is resolved I would have to turn it off to avoid throwing `eslint-disable-next-line` everywhere.

**Comment by j-a-y-h** [comment]
Alternatively, providing an option to the rule that relaxes these findings for that particular situation would be much appreciated as well.  Forgot to mention, thanks for maintaining this awesome library! :)
